hudi-spark-datasource/hudi-spark3.2plus-common/src/main/scala/org/apache/spark/sql/hudi/catalog/HoodieCatalog.scala

```diff
@@ -346,20 +346,25 @@ class HoodieCatalog extends DelegatingCatalogExtension
 }
 
 object HoodieCatalog {
-  def convertTransforms(partitions: Seq[Transform]): (Seq[String], Option[BucketSpec]) = {
+  def convertTransforms(transforms: Seq[Transform]): (Seq[String], Option[BucketSpec]) = {
     val identityCols = new mutable.ArrayBuffer[String]
     var bucketSpec = Option.empty[BucketSpec]
 
-    partitions.map {
+    transforms.foreach {
       case IdentityTransform(FieldReference(Seq(col))) =>
         identityCols += col
 
+      case MatchBucketTransform(numBuckets, col, sortCol) =>
```

Comment:
is this some bug fix. doesn't look like pure refactoring.

Reply:
This is fix for `BucketTransform` changed b/w Spark 3.2 and 3.3